nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java

```diff
@@ -464,6 +486,16 @@ private Properties getMailPropertiesFromFlowFile(final ProcessContext context, f
             }
         }
 
+        for (final PropertyDescriptor descriptor : context.getProperties().keySet()) {
+            if (descriptor.isDynamic()) {
+                final String flowFileValue = context.getProperty(descriptor).evaluateAttributeExpressions(flowFile).getValue();
+                // Nullable values are not allowed, so filter out
+                if (null != flowFileValue) {
+                    properties.setProperty(descriptor.getName(), flowFileValue);
```

Comment:
Recommend renaming `flowFileValue` to `mailPropertyValue`:
```suggestion
                final String mailPropertyValue = context.getProperty(descriptor).evaluateAttributeExpressions(flowFile).getValue();
                // Nullable values are not allowed, so filter out
                if (mailPropertyValue != null) {
                    properties.setProperty(descriptor.getName(), mailPropertyValue);
```

Reply:
Making the changes